src/Fhir.Anonymizer.AzureDataFactoryPipeline/src/FhirBlobConsumer.cs

```diff
@@ -0,0 +1,144 @@
+                await _currentStream.WriteAsync(byteData, 0, byteData.Length).ConfigureAwait(false);
+                await _currentStream.WriteAsync(_newLine, 0, _newLine.Length).ConfigureAwait(false);
+                if (_currentStream.Length >= UploadBlockThreshold)
+                {
+                    await AddCurrentStreamToUploadTaskListAsync().ConfigureAwait(false);
+                }
+
+                result++;
+            }
+
+            return result;
+        }
+
+        public async Task CompleteAsync()
+        {
+            if (_currentStream.Length > 0)
+            {
+                await AddCurrentStreamToUploadTaskListAsync().ConfigureAwait(false);
+            }
+
+            await Task.WhenAll(_runningTasks).ConfigureAwait(false);
+            foreach (var task in _runningTasks.FindAll(t => t.IsCompleted))
+            {
+                // If there's any error throw exception here.
+                await task.ConfigureAwait(false);
+            }
+
+            await _blobClient.CommitBlockListAsync(_blockIds).ConfigureAwait(false);
+        }
+
+        private async Task AddCurrentStreamToUploadTaskListAsync()
+        {
+            while (_runningTasks.Count > ConcurrentCount)
+            {
+                await Task.WhenAny(_runningTasks).ConfigureAwait(false);
+
+                foreach (var task in _runningTasks.FindAll(t => t.IsCompleted))
+                {
+                    // If there's any error throw exception here.
+                    await task.ConfigureAwait(false);
+                }
+
+                _runningTasks.RemoveAll(t => t.IsCompleted);
+            }
+
+            string blockId = GenerateBlockId(_blockIds.Count);
+            _runningTasks.Add(StageBlockAsync(blockId, _currentStream));
+            _blockIds.Add(blockId);
+
+            _currentStream = new MemoryStream();
+        }
+
+        private async Task StageBlockAsync(
+            string blockId,
+            MemoryStream stream)
+        {
+            await ExecutionWithTimeoutRetry.InvokeAsync<BlockInfo>(
+                async () =>
+                {
+                    using MemoryStream uploadStream = new MemoryStream();
+                    stream.Position = 0;
+                    await stream.CopyToAsync(uploadStream).ConfigureAwait(false);
+                    uploadStream.Position = 0;
+
+                    return await _blobClient.StageBlockAsync(blockId, uploadStream).ConfigureAwait(false);
+                }, TimeSpan.FromSeconds(BlockUploadTimeoutInSeconds), BlockUploadTimeoutRetryCount).ConfigureAwait(false);
+
+            ProgressHandler?.Report(stream.Length);
+            await stream.DisposeAsync().ConfigureAwait(false);
+        }
+
+        private static string GenerateBlockId(int index)
+        {
+            byte[] id = new byte[48]; // 48 raw bytes => 64 byte string once Base64 encoded
+            BitConverter.GetBytes(index).CopyTo(id, 0);
```

Comment:
not sure if i follow the reason for only 48 (<64) byte array

Reply:
It caused by the length limit for block is 64 bytes, and 48 bytes can be convert to a string with exactly 64 bytes.